### PR TITLE
Lite 21975 Redesign of EaaS websocket channel data models

### PR DESCRIPTION
**Pull request by bdjilka**
- skeleton of project
- new design of ws data models
- backwards capability (function to transform old schemas to new ones)

PR is open to discuss naming of new models, its attributes and nesting content.

### CODE REVIEW

File: demo.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# from connect.eaas.core.dataclasses import MessageType
```
Comment (ffaraoneim) [inline demo.py:1]: Please remove this file

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Topic :: Utilities",
+    "Topic :: Software Development :: Libraries",
+]
+
+[tool.poetry.dependencies]
+python = "^3.8"
+lark-parser = "0.11.0"
```
Comment (ffaraoneim) [inline pyproject.toml:27]: This dependency is not needed, please remove it

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+cachetools = "^5.0.0"
```
Comment (ffaraoneim) [inline pyproject.toml:28]: This one also, not needed, please remove

**Review by ffaraoneim — CHANGES_REQUESTED** [review]
Small changes are needed.
I miss a function to transform from v2 to v1 because we need it to respond to legacy runners.

File: connect/eaas/core/dataclasses.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    reply_to: Optional[str]
+
+
+class TaskInput(BaseModel):
+    event_type: str
+    object_id: str
+    data: Optional[Any]
+
+
+class TaskPayload(BaseModel):
```
[the remark this answers is not in the document]
Comment (bdjilka) [reply connect/eaas/core/dataclasses.py:72]: In my opinion, for consistency we should name all types of Message's model 'data' attribute the same way. So as 'Payload' in naming has place to be and plus/minus represent the meaning, maybe to leave 'TaskPayload'?

**Comment by bdjilka** [comment]
> I miss a function to transform from v2 to v1 because we need it to respond to legacy runners.

Ooops, missed it. Of course will implement.

File: connect/eaas/core/dataclasses.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    name: str
+    description: str
+
+
+class Repository(BaseModel):
+    readme_url: Optional[str]
+    changelog_url: Optional[str]
+
+
+class ExtensionPayload(BaseModel):
```
[the remark this answers is not in the document]
Comment (bdjilka) [reply connect/eaas/core/dataclasses.py:110]: Find it a bad naming :( Thought a lot about it, another variants were - 'ExtentionSetup', 'ExtentionConfiguration', simply 'Configuration' and 'Specification'

File: connect/eaas/core/dataclasses.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ffaraoneim) [reply connect/eaas/core/dataclasses.py:72]: Ok!

File: connect/eaas/core/dataclasses.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ffaraoneim) [reply connect/eaas/core/dataclasses.py:110]: Ok let it ExtensionPayload

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

**Pull request merged**